pkg/network/telemetry/metric_group.go

```diff
@@ -32,8 +32,12 @@ func NewMetricGroup(namespace string, commonTags ...string) *MetricGroup {
 
 // NewMetric returns a new `Metric` using the provided namespace and common tags
 func (mg *MetricGroup) NewMetric(name string, tags ...string) *Metric {
+	if mg.namespace != "" {
+		name = fmt.Sprintf("%s.%s", mg.namespace, name)
+	}
```

Comment:
I think you should add the condition to `Summary` method as well (where you are trying to trim the namespace prefix)

Reply:
There is no need to special case that (added a test to validate this)